osm2streets/src/render.rs

```diff
@@ -240,6 +251,65 @@ impl StreetNetwork {
         let output = serde_json::to_string_pretty(&obj)?;
         Ok(output)
     }
+
+    /// For an intersection, show all the movements.
+    pub fn debug_movements_geojson(&self) -> Result<String> {
+        let arrow_shift_dist = if self.config.driving_side == DrivingSide::Right {
+            Distance::meters(-1.3)
+        } else {
+            Distance::meters(1.3)
+        };
+
+        let mut pairs = Vec::new();
+
+        for (i, intersection) in &self.intersections {
+            // Find the points where the arrows should (leave, enter) the roads.
+            let road_points: BTreeMap<_, _> = intersection
```

Comment:
I wanted to implement a method on `Road` that could calculate these offset endpoints (for a given end and direction). Because a `Road` doesn't know it's own `DrivingSide` or dividing line (!!), it's non-trivial, so I'm happy with this currently.

I want to figure out how a `Road` should figure out it's own driving side, dividing line and `HalfRoad`s pretty soon. Maybe annotating contraflow lanes within `LaneSpec` would make it easy enough to calculate on demand from `lanes_lrt`, or maybe the driving side and dividing lane should be stored on the `Road` as part of the OSM parsing process (where we have the most context).

Reply:
I agree it's annoying how often `DrivingSide` has to be plumbed around. Just so I understand, by diving line, you mean the point where the direction of travel changes (or the left/right edge on one-way roads)? Looking at pairs of `lanes_ltr` and direction should work for that, right? And that process exposes cases where there isn't one obvious dividing lane (bidi cycletracks, shared center turn lanes)

> Maybe annotating contraflow lanes within LaneSpec

What do you mean by this? They have `Direction` already

Reply:
Yes, that is what I mean by "dividing line". If contraflow lanes were annotated as "contraflow" from the beginning, then we wouldn't get confused when trying to find the dividing line. Knowing the driving side might be enough to come up with a consistent definition / detection algorithm though.

Reply:
I'm open to trying either way. This is sounding like a good thing to unit test individually -- it's probably just a few weird edge cases I'm worrying about